ostd/src/trap/handler.rs

```diff
@@ -4,8 +4,9 @@ use core::sync::atomic::{AtomicBool, Ordering};
 
 use trapframe::TrapFrame;
 
-use crate::{arch::irq::IRQ_LIST, cpu_local};
+use crate::{arch::irq::IRQ_LIST, cpu_local, prelude::should_in_atomic};
 
+#[should_in_atomic]
 pub(crate) fn call_irq_callback_functions(trap_frame: &TrapFrame, irq_number: usize) {
```

Comment:
I am not convinced that annotating a function with `#[should_in_atomic]` is more superior than calling function `should_in_atomic` directly. The macro expansion is so simple that it is hardly more convenient than calling the function manually. Around 100 lines of code are added to implement the macros. But they won't be 100 call sites for the corresponding functions.